homepage/tests.py

```diff
@@ -103,3 +104,117 @@ def test_create_multiple_courses():
     # make sure that multiple courses are saved accurately
     courses = [Course.objects.get(pk=arg[0]).get_details() for arg in details]
     assert courses == details
+
+
+# -----------prerequisites tests----------- #
+
+# three courses:
+@pytest.fixture
+def courses():
+    courses_list = [
+         Course(1, "Course1", True, 3, "N/A", 1.3, 4, 5, 1),
+         Course(2, "Course2", False, 2, "N/A", 3.5, 2, 13, 5),
+         Course(3, "Course3", False, 1, "N/A", 2, 2.5, 0, 0)]
+    for course in courses_list:
+        course.save()
+
+    return courses_list
+
+
+# creating a new prerequisite log
+@pytest.mark.django_db
+def test_create_new_prerequisite(courses):
+    flag = True
+    preq = Prerequisites(course_id=courses[0], req_course_id=courses[1], req_code=Prerequisites.Req_Code.BEFORE)
+    try:
+        preq.clean_fields()
+    except ValidationError:
+        flag = False
+
+    assert flag
+
+    preq.save()
+
+    assert Prerequisites.objects.filter(pk=preq.id).exists()
+    assert Prerequisites.objects.filter(pk=preq.id).count() == 1

```

Comment:
I think it's a bit redundant since the DB won't allow creation of two Preq' with the same PK.
Maybe assert Prerequisites.objects.filter(course_id=courses[0]).count() == 1 to check that only one row was created for this course would be better, or just leave the first assert.

Reply:
yeah, I messed this up, I intended to check for the course, will change